packages/astro/src/runtime/server/transition.ts

```diff
@@ -45,6 +45,29 @@ const addPairs = (
 	}
 };
 
+// Chrome (121) accepts custom-idents for view-transition-names as generated by cssesc,
+// but it just ignores them during view transitions if they contain escaped 7-bit ASCII characters
+// like \<space> or \. A special case are digits and minus at the beginning of the string,
+// which cssesc also encodes as \xx
+const reEncodeValidChars: string[] = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-_"
+	.split('').reduce((v, c) => (v[c.charCodeAt(0)] = c, v), [] as string[]);
+const reEncodeInValidStart: string[] = "-0123456789"
+	.split('').reduce((v, c) => (v[c.charCodeAt(0)] = c, v), [] as string[]);
+
+function reEncode(s: string) {
```

Comment:
Can you add some comments that explain the re-encoding logic?

Reply:
Sure, I can't understand why I didn't do that from the start ;-)